ansible/os-uod-docker.yml

```diff
@@ -0,0 +1,95 @@
+---
+# Playbook for starting a docker instance in the
+# openstack cloud.
+
+## See os-create.yml
+
+- hosts: localhost
+  connection: local
+  #gather_facts: false
+
+  vars:
+
+    vm_image: "CentOS 7 1604"
+    vm_flavour: m2.large
+    vm_groups: "ansible-managed,os-image-centos,docker-hosts"
+    vm_extra_groups: ""
+    ignore_internal_known_hosts: True
+
+  pre_tasks:
+
+    - fail:
+        msg: "vm_key_name is required"
+      when: vm_key_name is undefined or not vm_key_name
+
+    - fail:
+        msg: "vm_name is required"
+      when: vm_name is undefined or not vm_name
+
+
+  tasks:
+
+    - set_fact: vm_joined_groups="{{ vm_groups.split(',') | union(vm_extra_groups.split(',')) | join(',')}}"
+
+    - name: Docker external access security group
+      os_security_group:
+        description: External access to Docker servers (managed by Ansible)
+        name: docker-external
+        state: present
+
+    - name: Docker external access security group rules
+      os_security_group_rule:
+        direction: ingress
+        port_range_max: "{{ item }}"
+        port_range_min: "{{ item }}"
+        protocol: tcp
+        remote_ip_prefix: 0.0.0.0/0
+        security_group: docker-external
+        state: present
+      with_items:
+        - 22
+        - 80
+        - 443
+        - 4063
+        - 4064
+        - 8008
+        - 8888
+
+    - name: Docker VM
+      os_server:
+        name: "{{ vm_name }}"
+        state: present
+        image: "{{ vm_image }}"
+        key_name: "{{ vm_key_name }}"
+        flavor: "{{ vm_flavour }}"
+        auto_ip: yes
+        meta:
+          hostname: "{{ vm_name }}"
+          groups: "{{ vm_joined_groups }}"
+        security_groups: [default, docker-external]
+      register: vmdocker
+
+    # To prevent duplicates with the dynamic inventory only add_host if
+    # the VM was created in this run
+
+    - name: Docker VM hosts files
+      add_host:
+        name: "{{ vm_name }}"
+        groups: "{{ vm_joined_groups }}"
+        ansible_host: "{{ vmdocker.server.public_v4 }}"
+        ansible_user: "centos"
+        ansible_become: true
+      when: vmdocker.changed
```

Comment:
Is this copied from the IDR playbooks? This is only needed if you want to create VMs and deploy servers in the same playbook, and is one of the things we discussed regarding unnecessary complexity with the IDR deployments. Just wanted to double-check you're happy with that here (it's self contained, so not such a big problem).